spec/std/system_spec.cr

```diff
@@ -4,16 +4,21 @@ require "system"
 describe System do
   describe "hostname" do
     it "returns current hostname" do
-      shell_hostname = `hostname`.strip
-      $?.success?.should be_true # The hostname command has to be available
+      shell_hostname = {{ `hostname`.stringify }}.strip # Workaround for inability to execute shell commands on Windows.
+      $?.success?.should be_true                        # The hostname command has to be available
       hostname = System.hostname
       hostname.should eq(shell_hostname)
     end
   end
 
   describe "cpu_count" do
     it "returns current CPU count" do
-      shell_cpus = `getconf _NPROCESSORS_ONLN 2>/dev/null || nproc --all 2>/dev/null || grep -sc '^processor' /proc/cpuinfo || sysctl -n hw.ncpu 2>/dev/null`.to_i
+      shell_cpus = 0
```

Comment:
This line seems unnecessary to me.

Reply:
If I don't declare it, the equality check `cpu_count.should eq(shell_cpus)` fails because it can't find a declaration for the variable.

Reply:
That's odd because it should be defined in all macro branches.

Reply:
You should be able to remove this line.

Reply:
[CI](https://circleci.com/gh/crystal-lang/crystal/15436?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) fails with that removed.

Reply:
```cr
shell_cpus = {% if ...
```

Reply:
Ah, I didn't know that you could conditionally define stuff using macros.